src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -13685,6 +13684,298 @@ instruct SubExtI_uxth_and_shift(iRegINoSp dst, iRegIorL2I src1, iRegIorL2I src2,
+// DO NOT EDIT ANYTHING IN THIS SECTION OF THE FILE
+instruct cmovI_reg_imm0_lt(iRegINoSp dst, iRegI src1, rFlagsReg cr)
+%{
+  effect(DEF dst, USE src1, USE cr);
+  ins_cost(INSN_COST * 2);
+  format %{ "cselw $dst, $src1, zr lt\t"  %}
+
+  ins_encode %{
+    __ cselw($dst$$Register,
+             $src1$$Register,
+             zr,
+             Assembler::LT);
+  %}
+  ins_pipe(icond_reg);
+%}
+
+// This pattern is automatically generated from aarch64_ad.m4
+// DO NOT EDIT ANYTHING IN THIS SECTION OF THE FILE
+instruct cmovI_reg_imm0_gt(iRegINoSp dst, iRegI src1, rFlagsReg cr)
+%{
+  effect(DEF dst, USE src1, USE cr);
+  ins_cost(INSN_COST * 2);
+  format %{ "cselw $dst, $src1, zr gt\t"  %}
+
+  ins_encode %{
+    __ cselw($dst$$Register,
+             $src1$$Register,
+             zr,
+             Assembler::GT);
+  %}
+  ins_pipe(icond_reg);
+%}
+
+// This pattern is automatically generated from aarch64_ad.m4
+// DO NOT EDIT ANYTHING IN THIS SECTION OF THE FILE
+instruct cmovI_reg_imm1_le(iRegINoSp dst, iRegI src1, rFlagsReg cr)
+%{
+  effect(DEF dst, USE src1, USE cr);
+  ins_cost(INSN_COST * 2);
+  format %{ "csincw $dst, $src1, zr le\t"  %}
+
+  ins_encode %{
+    __ csincw($dst$$Register,
+             $src1$$Register,
+             zr,
+             Assembler::LE);
+  %}
+  ins_pipe(icond_reg);
+%}
+
+// This pattern is automatically generated from aarch64_ad.m4
+// DO NOT EDIT ANYTHING IN THIS SECTION OF THE FILE
+instruct cmovI_reg_imm1_gt(iRegINoSp dst, iRegI src1, rFlagsReg cr)
+%{
+  effect(DEF dst, USE src1, USE cr);
+  ins_cost(INSN_COST * 2);
+  format %{ "csincw $dst, $src1, zr gt\t"  %}
+
+  ins_encode %{
+    __ csincw($dst$$Register,
+             $src1$$Register,
+             zr,
+             Assembler::GT);
+  %}
+  ins_pipe(icond_reg);
+%}
+
+// This pattern is automatically generated from aarch64_ad.m4
+// DO NOT EDIT ANYTHING IN THIS SECTION OF THE FILE
+instruct cmovI_reg_immM1_lt(iRegINoSp dst, iRegI src1, rFlagsReg cr)
+%{
+  effect(DEF dst, USE src1, USE cr);
+  ins_cost(INSN_COST * 2);
+  format %{ "csinvw $dst, $src1, zr lt\t"  %}
+
+  ins_encode %{
+    __ csinvw($dst$$Register,
+             $src1$$Register,
+             zr,
+             Assembler::LT);
+  %}
+  ins_pipe(icond_reg);
+%}
+
+// This pattern is automatically generated from aarch64_ad.m4
+// DO NOT EDIT ANYTHING IN THIS SECTION OF THE FILE
+instruct cmovI_reg_immM1_ge(iRegINoSp dst, iRegI src1, rFlagsReg cr)
+%{
+  effect(DEF dst, USE src1, USE cr);
+  ins_cost(INSN_COST * 2);
+  format %{ "csinvw $dst, $src1, zr ge\t"  %}
+
+  ins_encode %{
+    __ csinvw($dst$$Register,
+             $src1$$Register,
+             zr,
+             Assembler::GE);
+  %}
+  ins_pipe(icond_reg);
+%}
+
+// This pattern is automatically generated from aarch64_ad.m4
+// DO NOT EDIT ANYTHING IN THIS SECTION OF THE FILE
+instruct minI_reg_imm0(iRegINoSp dst, iRegIorL2I src, immI0 imm)
+%{
+  match(Set dst (MinI src imm));
+  ins_cost(INSN_COST * 2);
```

Comment:
I think the cost of the node before expansion should be the sum of the nodes expanded. So the cost here and other places should be INSN_COST * 3 instead.

Reply:
Agree, corrected in a new version.